src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -485,6 +503,163 @@ private static GapicProductConfig createDummyInstance(
             .collect(Collectors.toMap(InterfaceConfigProto::getName, Function.identity())));
   }
 
+  static List<InterfaceConfigProto> injectRetryPolicyConfig(
```

Comment:
Hmmmm, injecting retry configs back to gapic config seems a bit counter-intuitive. This means we can't delete those fields from gapic_config proto, even though after this change, we do not anticipate users to continue to configure the retry fields through gapic config (if that is true).

Just thinking out loud, is it possible to inject retry policy configs directly to `MethodConfig`?

Reply:
> This means we can't delete those fields from gapic_config proto

Ah I see your concern here.

>  is it possible to inject retry policy configs directly to MethodConfig?

Mmm I'm not following (my own unfamiliarity with the code base), where does the MethodConfig come into play during execution? Also, where is it defined? (If you have time, links would be helpful :smiley: )

The reason I went with essentially "spoofing" the GAPIC config is because there are N (unknown to me, might be known to you) downstream things that languages do with the GAPIC config after receiving it for generation. Instead of wading through that, I figured we could inject it with the new values and languages could do with them as they see fit.

Reply:
_(I figured that was the reason and I totally agree gapic-generator is super complicated)_

Correction: instead of injecting directly to [`MethodConfig`](https://github.com/googleapis/gapic-generator/blob/46f49e621c5aa04f03afcbf4abc7391648ab6ab7/src/main/java/com/google/api/codegen/config/MethodConfig.java#L49-L51), the right way to do this might be injecting retry configs to [`GapicInterfaceConfig`](https://github.com/googleapis/gapic-generator/blob/46f49e621c5aa04f03afcbf4abc7391648ab6ab7/src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java#L73-L77). And GapicInterfaceConfig would most likely pass the along to `MethodConfig`.

I think gapic-generator has this intermediate layer ([usually ***Config](https://github.com/googleapis/gapic-generator/tree/master/src/main/java/com/google/api/codegen/config)) that holds somewhat preprocessed information from gapic config yamls. (I don't exactly know why, but I guess the design was there for a bunch of reasons like being less proto-ish and more Java idiomatic, to support discogapics, to make config schema change easier etc). [`GapicProductConfig`](https://github.com/googleapis/gapic-generator/blob/master/src/main/java/com/google/api/codegen/config/GapicProductConfig.java) is the top-level Config object, and is the starting point for gapic generations.

To be more specific, retry codes are re-written into [`RetryCodesConfig`](https://github.com/googleapis/gapic-generator/blob/master/src/main/java/com/google/api/codegen/config/RetryCodesConfig.java), but 
retry setting definitions are not re-written into an intermediate object (we are using [protos](https://github.com/googleapis/gapic-generator/blob/46f49e621c5aa04f03afcbf4abc7391648ab6ab7/src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java#L77) directly, which starts to be annoying), and they are created and stored in `GapicInterfaceConfig` [here](https://github.com/googleapis/gapic-generator/blob/46f49e621c5aa04f03afcbf4abc7391648ab6ab7/src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java#L120-L152).

So instead of injecting into the config proto objects directly, probably alter the creation of `retryCodeConfig` and `retrySettingDefinition` mentioned in the above link and create those two from grpc service config protos?

Note, Ideally I think we should also change the type of [`retrySettingsDefinition`](https://github.com/googleapis/gapic-generator/blob/46f49e621c5aa04f03afcbf4abc7391648ab6ab7/src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java#L77) to some intermediate config objects and not depend on the protos (so that we can completely delete those fields from gapic config), but that would cause huge refactoring and probably should be done separately.


FYI the pipeline works like:
```
yamls -> parsed into protos -> re-written into the ***Config objects -> put into contexts
-> passed to transformers -> ... -> rendering
```

Reply:
Ok this is great! Thank you for the guidance. Everything looks like it'll still be in a similar place, I've just gotta refactor the logic a bit. The goal of removing the retry fields from the proto entirely is a great one, good call.

Reply:
Ehh it might be more work than I first anticipated. I'll lyk when to take a look, might be a few commits coming in.